opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -201,15 +201,26 @@ func MapClusterRole(role string, ver string) string {
 	if err != nil {
 		return role
 	}
-	clusterManagerVer, _ := version.NewVersion("2.0.0")
-	is2XVersion := osVer.GreaterThanOrEqual(clusterManagerVer)
-	if role == "master" && is2XVersion {
-		return "cluster_manager"
-	} else if role == "cluster_manager" && !is2XVersion {
-		return "master"
-	} else {
-		return role
+
+	majorVersion := osVer.Segments()[0]
```

Comment:
Good one @tomsnuvv, initially I was confused if it has to anything with OpenSearch segments, but this `Segments` is coming from `github.com/hashicorp/go-version`.